src/game/server.cpp

```diff
@@ -3348,7 +3348,11 @@ namespace server
                 if(i == SENDMAP_MPZ) smapcrc = crcfile(reqfile);
                 mapdata[i] = openfile(reqfile, "rb");
             }
-            if(!hasmapdata()) resetmapdata();
+            if(!hasmapdata())
+            {
+                logoutf("Could not load requested map: %s", reqmap);
```

Comment:
Is this message shown not only when the server cannot find a map locally on the filesystem but none of the clients can upload one either?

Reply:
This message is shown when the server itself cannot load the map; clients may still provide it themselves.

Reply:
I see. Do you happen to know where we'd have to add another improved error message? Also, I'd suggest to reword this message to clearly communicate that the server couldn't find the map itself, and that it'll attempt to upload it from one of the clients and distribute it to the others.

Reply:
Reworded.

Adding a message for when clients haven't got the map is harder: a client's map sending can be delayed for whatever reason.